diff --git a/node_modules/moti/build/interactions/pressable/pressable.js b/node_modules/moti/build/interactions/pressable/pressable.js
index c25d475..b1b0d2e 100644
--- a/node_modules/moti/build/interactions/pressable/pressable.js
+++ b/node_modules/moti/build/interactions/pressable/pressable.js
@@ -1,6 +1,6 @@
 import { jsx as _jsx } from "react/jsx-runtime";
 import { useMemo, forwardRef } from 'react';
-import { Pressable } from 'react-native';
+import { Pressable } from 'react-native-gesture-handler';
 import { useSharedValue, runOnJS, useDerivedValue, } from 'react-native-reanimated';
 import { View as MotiView } from '../../components/view';
 import { MotiPressableContext, useMotiPressableContext, INTERACTION_CONTAINER_ID, } from './context';
diff --git a/node_modules/moti/src/interactions/pressable/pressable.tsx b/node_modules/moti/src/interactions/pressable/pressable.tsx
index 2873b45..1bef4f9 100644
--- a/node_modules/moti/src/interactions/pressable/pressable.tsx
+++ b/node_modules/moti/src/interactions/pressable/pressable.tsx
@@ -1,5 +1,5 @@
 import { useMemo, ReactNode, forwardRef } from 'react'
-import { Pressable } from 'react-native'
+import { Pressable } from 'react-native-gesture-handler'
 import type { View } from 'react-native'
 import {
   useSharedValue,
