diff --git a/node_modules/react-native-gesture-handler/ios/RNGestureHandlerEvents.m b/node_modules/react-native-gesture-handler/ios/RNGestureHandlerEvents.m
index 6e04f51..c4ae2ae 100644
--- a/node_modules/react-native-gesture-handler/ios/RNGestureHandlerEvents.m
+++ b/node_modules/react-native-gesture-handler/ios/RNGestureHandlerEvents.m
@@ -12,6 +12,13 @@ - (instancetype)initWithData:(NSDictionary *)data;
   return self;
 }
 
++ (uint64_t)currentTimestampInMilliseconds {
+    NSTimeInterval unixEpochTimeInSeconds = [[NSDate date] timeIntervalSince1970];
+    uint64_t unixEpochTimeInMilliseconds = (uint64_t)(unixEpochTimeInSeconds * 1000);
+
+    return unixEpochTimeInMilliseconds;
+}
+
 + (RNGestureHandlerEventExtraData *)forPosition:(CGPoint)position withAbsolutePosition:(CGPoint)absolutePosition
 {
   return [[RNGestureHandlerEventExtraData alloc] initWithData:@{
@@ -56,6 +63,8 @@ + (RNGestureHandlerEventExtraData *)forPan:(CGPoint)position
                               withVelocity:(CGPoint)velocity
                        withNumberOfTouches:(NSUInteger)numberOfTouches
 {
+  uint64_t timestamp = [self currentTimestampInMilliseconds];
+
   return [[RNGestureHandlerEventExtraData alloc] initWithData:@{
     @"x" : @(position.x),
     @"y" : @(position.y),
@@ -65,7 +74,8 @@ + (RNGestureHandlerEventExtraData *)forPan:(CGPoint)position
     @"translationY" : @(translation.y),
     @"velocityX" : SAFE_VELOCITY(velocity.x),
     @"velocityY" : SAFE_VELOCITY(velocity.y),
-    @"numberOfPointers" : @(numberOfTouches)
+    @"numberOfPointers" : @(numberOfTouches),
+    @"timestamp": @(timestamp)
   }];
 }
 
diff --git a/node_modules/react-native-gesture-handler/lib/typescript/handlers/PanGestureHandler.d.ts b/node_modules/react-native-gesture-handler/lib/typescript/handlers/PanGestureHandler.d.ts
index 4685c5b..7f628e6 100644
--- a/node_modules/react-native-gesture-handler/lib/typescript/handlers/PanGestureHandler.d.ts
+++ b/node_modules/react-native-gesture-handler/lib/typescript/handlers/PanGestureHandler.d.ts
@@ -50,6 +50,10 @@ export type PanGestureHandlerEventPayload = {
      * value is expressed in point units per second.
      */
     velocityY: number;
+    /**
+     * The number of milliseconds elapsed since the epoch of the pan gesture in the current moment.
+     */
+    timestamp: number;
 };
 interface CommonPanProperties {
     /**
